tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -146,11 +118,37 @@ def preprocess_train(self, X_train, y_train, X_val, y_val, **kwargs):
                 .databunch(bs=self.params['bs'] if len(X_train) > self.params['bs'] else 32, num_workers=num_workers))
         return data
 
-    def fill_missing(self, df):
-        df = df[self.cat_columns + self.cont_columns].copy()
-        for c in self.cat_columns + self.cont_columns:
-            if c in self.cat_columns:
-                df[c] = df[c].cat.add_categories(MISSING)
+    def _preprocess(self, X: pd.DataFrame, fit=False, **kwargs):
+        X = super()._preprocess(X=X, **kwargs)
+        if fit:
```

Comment:
it is confusing to have `_preprocess_train` and this `if fit` here especially because lots of logic is hidden in abstract class: hard to read what happening when.